File: Ex2Num8.swift
import Foundation

print("Введите количество цветов: ", terminator: "")
guard let nInput = readLine(), let n = Int(nInput) else {
    print("Некорректный ввод.")
    exit(0)
}

print("Введите потребность каждого цветка через пробел:")
guard let waterInput = readLine() else {
    print("Некорректный ввод.")
    exit(0)
}

let waterNeeds = waterInput.split(separator: " ").compactMap { Int($0) }
if waterNeeds.count != n {
    print("Количество введённых потребностей не совпадает с количеством цветов.")
    exit(0)
}

print("Введите объем лейки: ", terminator: "")
guard let capacityInput = readLine(), let capacity = Int(capacityInput) else {
    print("Некорректный ввод.")
    exit(0)
}

var steps = 0
var currentWater = capacity
var position = -1 // начинаем с реки

for i in 0..<n {
    if currentWater >= waterNeeds[i] {
        steps += abs(position - i)
        currentWater -= waterNeeds[i]
        position = i
    } else {
        // идем к реке
        steps += abs(position - (-1))
        position = -1
        currentWater = capacity

        // обратно к цветку
        steps += abs(position - i)
        currentWater -= waterNeeds[i]
        position = i
    }
}

print("Общее количество шагов: \(steps)")

File: Ex2Num8.js
const readline = require('readline-sync');

const n = parseInt(readline.question("Введите количество цветов: "));
const waterNeeds = readline.question("Введите потребность каждого цветка через пробел: ")
                         .split(' ')
                         .map(Number);
const capacity = parseInt(readline.question("Введите объем лейки: "));

let steps = 0;
let currentWater = capacity;
let position = -1; // начинаем с реки

for (let i = 0; i < n; i++) {
  if (currentWater >= waterNeeds[i]) {
    steps += Math.abs(position - i);
    currentWater -= waterNeeds[i];
    position = i;
  } else {
    steps += Math.abs(position - (-1)); // до реки
    position = -1;
    currentWater = capacity;

    steps += Math.abs(position - i); // обратно к цветку
    currentWater -= waterNeeds[i];
    position = i;
  }
}

console.log("Общее количество шагов: " + steps);

